src/daterangepicker/components/date-range-picker.component.html

```diff
@@ -9,88 +9,429 @@
+									[class]="calendarVariables.right.classes[row][col]"
+									(click)="clickDate($event, sideEnum.right, row, col)"
+									[attr.data-cy]="'date-range-picker-row-' + rowIndex + '-col-' + colIndex + '__button'"
+								>
+									<span>{{ calendarVariables.right.calendar[row][col].date() }}</span>
+								</td>
+							</tr>
+						</tbody>
+					</table>
+				</div>
+				<div class="calendar-time" *ngIf="timePicker">
+					<div class="select">
+						<select
+							class="select-item hourselect"
+							[disabled]="!endDate"
+							[(ngModel)]="timepickerVariables.right.selectedHour"
+							(ngModelChange)="timeChanged($event, sideEnum.right)"
+						>
+							<option
+								*ngFor="let i of timepickerVariables.right.hours"
+								[value]="i"
+								[disabled]="timepickerVariables.right.disabledHours.indexOf(i) > -1"
+								>{{ i }}</option
+							>
+						</select>
+						<span class="select-highlight"></span>
+						<span class="select-bar"></span>
+					</div>
+					<div class="select">
+						<select
+							class="select-item minuteselect"
+							[disabled]="!endDate"
+							[(ngModel)]="timepickerVariables.right.selectedMinute"
+							(ngModelChange)="timeChanged($event, sideEnum.right)"
+						>
+							<option
+								*ngFor="let i of timepickerVariables.right.minutes; let index = index"
+								[value]="i"
+								[disabled]="timepickerVariables.right.disabledMinutes.indexOf(i) > -1"
+								>{{ timepickerVariables.right.minutesLabel[index] }}</option
+							>
+						</select>
+						<span class="select-highlight"></span>
+						<span class="select-bar"></span>
+					</div>
+					<div class="select">
+						<select
+							*ngIf="timePickerSeconds"
+							class="select-item secondselect"
+							[disabled]="!endDate"
+							[(ngModel)]="timepickerVariables.right.selectedSecond"
+							(ngModelChange)="timeChanged($event, sideEnum.right)"
+						>
+							<option
+								*ngFor="let i of timepickerVariables.right.seconds; let index = index"
+								[value]="i"
+								[disabled]="timepickerVariables.right.disabledSeconds.indexOf(i) > -1"
+								>{{ timepickerVariables.right.secondsLabel[index] }}</option
+							>
+						</select>
+						<span class="select-highlight"></span>
+						<span class="select-bar"></span>
+					</div>
+					<div class="select">
+						<select
+							*ngIf="!timePicker24Hour"
+							class="select-item ampmselect"
+							[(ngModel)]="timepickerVariables.right.ampmModel"
+							(ngModelChange)="timeChanged($event, sideEnum.right)"
+						>
+							<option value="AM" [disabled]="timepickerVariables.right.amDisabled">AM</option>
+							<option value="PM" [disabled]="timepickerVariables.right.pmDisabled">PM</option>
+						</select>
+						<span class="select-highlight"></span>
+						<span class="select-bar"></span>
+					</div>
+				</div>
+			</div>
+		</div>
+		<div class="dp-footer" *ngIf="!autoApply && (!ranges.length || (showCalInRanges && !singleDatePicker))">
```

Comment:
The footer is disappearing on a small responsive layout when the height is below 740px so if we deferred the future scroll work, this still isn't a usable solution.  We would want two calendars stacked on top of each other not to push the action buttons in the footer out of the way.

I'm thinking we can mirror the same structure from PS's small layout in this component.

here's a codepen that isolates the positioning styles:
https://codepen.io/jefferywroblewski-accesso/pen/ZEGrogB

We would need to wrap the whole thing in a new container element.  Then you have to wrap the body and footer in another container to get everything in the right place with the scrollable body.
Give that a go and see if the existing contents here just fits in the flex containers.

Reply:
This is a very good catch, I applied your suggestions and it worked perfectly.  One caveat, something in the demo of this control is not playing nicely, like a width not being set somewhere, so the width of the picker gets a bit wonky, but it works perfectly in Passport Support.